File: app/src/main/java/com/music/player/bhandari/m/model/TrackItem.java
package com.music.player.bhandari.m.model;

import android.media.MediaMetadataRetriever;
import android.util.Log;

import java.io.Serializable;

/**
 * Created by amit on 29/11/16.
 */

public class TrackItem implements Serializable {
    private int id;
    private String filePath="";
    private String title="";
    private String artist="";
    private String album="";
    private String genre="";
    private String duration="";  //string in milliseconds
    private int album_id;
    private int artist_id;
    //default constructor
    public TrackItem() {

    }

    public TrackItem(String filePath, String title, String artist, String album,String genre, String duration
            , int album_id, int artist_id, int id){
        this.filePath=filePath;
        this.title=title;
        this.artist=artist;
        this.album=album;
        this.genre=genre;
        this.duration=duration;
        this.album_id=album_id;
        this.id = id;
        this.artist_id=artist_id;
    }



    //if needed in case
    public TrackItem(String filePath) {
        MediaMetadataRetriever mmr = new MediaMetadataRetriever();
        Log.e("filepath",filePath);
        mmr.setDataSource(filePath);
        String duration=mmr.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION);
        this.title=(mmr.extractMetadata(MediaMetadataRetriever.METADATA_KEY_TITLE));
        this.artist=(mmr.extractMetadata(MediaMetadataRetriever.METADATA_KEY_ARTIST));
        this.album=(mmr.extractMetadata(MediaMetadataRetriever.METADATA_KEY_ALBUM));
        this.duration=(mmr.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION));
        this.filePath=filePath;
    }

    //return in ms
    public int getDurInt(){
        int durationMs = 0;
        try {
            durationMs = Integer.parseInt(duration);
        }catch (NumberFormatException e){
            //send predefined number 300 msec
            durationMs = 3000;
        }
        return durationMs;
    }

    public String getDurStr(){
        int minutes=(Integer.parseInt(duration)/1000)/60;
        int seconds= (Integer.parseInt(duration)/1000)%60;
        String durFormatted=String.format("%02d",seconds);
        return  String.valueOf(minutes)+":"+durFormatted;
    }

    public void setId(int id){
        this.id = id;
    }

    public int getId() {
        return id;
    }

    public String getFilePath() {
        return filePath;
    }

    public void setFilePath(String title) {
        this.filePath = filePath;
    }

    public String getGenre() {
        return genre;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getAlbum() {
        return album;
    }

    public void setAlbum(String title) {
        this.album = title;
    }

    public String getArtist() {
        return artist;
    }

    public void setArtist(String artist) {
        if (artist==null || artist.isEmpty()){
            this.artist="<unknown>";
            return;
        }
        this.artist = artist;
    }

    public String getDuration() {
        return duration;
    }

    public void setDuration(String duration) {
        this.duration = duration;
    }

    public int getAlbumId(){return album_id;}

    public int getArtist_id(){return artist_id;}

    public void setArtist_id(int artist_id){
        this.artist_id = artist_id;
    }

    public boolean haveAlbumArt(){
        return MusicLibrary.getInstance().getAlbumArtUri(album_id) != null;
    }
}

File: app/src/main/java/com/music/player/bhandari/m/model/TrackItem.kt
package com.music.player.bhandari.m.model

import android.media.MediaMetadataRetriever
import android.util.Log
import java.io.Serializable

/**
 * Copyright 2017 Amit Bhandari AB
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
class TrackItem : Serializable {
    var id = 0
    private var filePath = ""
    var title: String? = ""
    var artist: String? = ""
    var album: String? = ""
    var genre = ""
        private set
    var duration: String? = "" //string in milliseconds
    var albumId = 0
        private set
    var artist_id = 0

    //default constructor
    constructor()
    constructor(
        filePath: String,
        title: String?,
        artist: String?,
        album: String?,
        genre: String,
        duration: String?,
        album_id: Int,
        artist_id: Int,
        id: Int
    ) {
        this.filePath = filePath
        this.title = title
        this.artist = artist
        this.album = album
        this.genre = genre
        this.duration = duration
        albumId = album_id
        this.id = id
        this.artist_id = artist_id
    }

    //if needed in case
    constructor(filePath: String) {
        val mmr = MediaMetadataRetriever()
        Log.e("filepath", filePath)
        mmr.setDataSource(filePath)
        val duration = mmr.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION)
        title = mmr.extractMetadata(MediaMetadataRetriever.METADATA_KEY_TITLE)
        artist = mmr.extractMetadata(MediaMetadataRetriever.METADATA_KEY_ARTIST)
        album = mmr.extractMetadata(MediaMetadataRetriever.METADATA_KEY_ALBUM)
        this.duration = mmr.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION)
        this.filePath = filePath
    }//send predefined number 300 msec

    //return in ms
    val durInt: Int
        get() {
            var durationMs = 0
            durationMs = try {
                duration!!.toInt()
            } catch (e: NumberFormatException) {
                //send predefined number 300 msec
                3000
            }
            return durationMs
        }
    val durStr: String
        get() {
            val minutes = duration!!.toInt() / 1000 / 60
            val seconds = duration!!.toInt() / 1000 % 60
            val durFormatted = String.format("%02d", seconds)
            return "$minutes:$durFormatted"
        }

    fun getFilePath(): String {
        return filePath
    }

    fun setFilePath(title: String?) {
        filePath = filePath
    }

    @JvmName("getArtist1")
    fun getArtist(): String? {
        return artist
    }

    @JvmName("setArtist1")
    fun setArtist(artist: String?) {
        if (artist == null || artist.isEmpty()) {
            this.artist = "<unknown>"
            return
        }
        this.artist = artist
    }

    fun haveAlbumArt(): Boolean {
        return MusicLibrary.instance.getAlbumArtUri(albumId) != null
    }
}